src/core/thread.d

```diff
@@ -631,7 +633,8 @@ class Thread
         }
 
         // Start thread as non-suspendable, undone in thread_entryPoint
-        atomicStore(m_isInCriticalRegion, true);
+        synchronized (Thread.criticalRegionLock)
+            m_isInCriticalRegion = true;
```

Comment:
So thread_suspendAll wouldn't work when being called from DllMain. As said before we should do as little as possible in DllMain.
I'll work out a different solution anyhow, the real problem is, that we're adding threads to the global list before they are started.